### PR TITLE
Lint and indentation fixes

### PR DESCRIPTION
**Pull request by dpvc**
This PR makes changes to remove tslint problems, compiler problems (unused variables, etc.), and indentation problems (moves to 2 rather than 4 characters).

There are changes to `tsconfig.json` and `tslint.json` (see the initial commit) to add stricter rules for compiling, but lose the positioning rules so that public/protected/private can be intermixed (as per F2F).

The second commit is the tslint and compiler changes.  These did require some code to be moved, but most of the rest was adding jsdoc comments and some function return types.  For unused variables in function argument lists, the solution is to use a name starting with an underscore (that disables the warning).

It would be good to go through and make sure all the jsdoc actually reflects the functions themselves, but I didn't do that.  We also should mark the optional and initialized arguments, which often isn't the case in the jsdoc.  But I think I've done enough for now.

The third commit does the indentation change.  A few lines needed to be broken differently to reduce the resulting automatic indentation of following lines.

I don't think there is really any need to review this carefully, as it is all cosmetic.  Looking at the commits separately might be easier than looking at the final result, if you do want to check anything.

### CODE REVIEW

**Comment by dpvc** [comment]
Note that you can hide white-space changed in the menu under the gear icon in the file list.

**Comment by dpvc** [comment]
This accidentally included the fix for issue mathjax/MathJax#2412.  It is only one line, and a number of other branches are based off this one, so I'm leaving it here.  Sorry about that.

File: ts/components/startup.ts [hunk 1]
```diff
@@ -120,380 +122,380 @@ declare var global: {document: Document};
...
-        return mathjax.document(root || CONFIG.document, {
-            ...MathJax.config.options,
-            InputJax: input,
-            OutputJax: output
-        });
+    MathJax.typesetClear = () => document.clear();
+  }
+
+  /**
+   * Make the input2output(math, options?) and input2outuputPromise(math, options?) methods,
```
Comment (zorkow) [inline ts/components/startup.ts:346]: Typo

File: ts/core/InputJax.ts [hunk 2]
```diff
@@ -114,88 +114,115 @@ export interface InputJax<N, T, D> {
...
+   * The DOMAdaptor for the MathDocument for this input jax
+   */
+  public adaptor: DOMAdaptor<N, T, D> = null;  // set by the handler
+  /**
+   * The MathML node factory
+   */
+  public mmlFactory: MmlFactory = null;        // set by the handler
+
+  /**
+   * @param {OptionList} options  The options to applyt to this input jax
```
Comment (zorkow) [inline ts/core/InputJax.ts:152]: typo

File: ts/core/InputJax.ts [hunk 3]
```diff
@@ -114,88 +114,115 @@ export interface InputJax<N, T, D> {
...
+   */
+  public abstract compile(math: MathItem<N, T, D>, document: MathDocument<N, T, D>): MmlNode;
+
+  /**
+   * Execute a set of filters, passing them the MathItem and any needed data,
+   *  and return the (possibly modified) data
+   *
+   * @param {FunctionList} filters   The list of functions to be performed
+   * @param {MathItem} math          The math item that is being processed
+   * @param {MathDocument} document  The math document contaiing the math item
```
Comment (zorkow) [inline ts/core/InputJax.ts:215]: typo

File: ts/util/FunctionList.ts [hunk 4]
```diff
@@ -37,60 +37,61 @@ export interface FunctionListItem extends PrioritizedListItem<Function> {}
...
+      let result = item.item(...data);
+      if (result === false) {
+        return false;
+      }
     }
+    return true;
+  }
+
+  /**
+   * Executes the functions in the list (in prioritied order) asynchronously,
```
Comment (zorkow) [inline ts/util/FunctionList.ts:60]: typo

File: ts/core/MathDocument.ts [hunk 5]
```diff
@@ -254,178 +254,178 @@ export class RenderList<N, T, D> extends PrioritizedList<RenderData<N, T, D>> {
...
+   */
+  options: OptionList;
+
+  /**
+   * The list of MathItems found in this page
+   */
+  math: MathList<N, T, D>;
+
+  /**
+   * The list of actions to take durring a render() or convert() call
```
Comment (zorkow) [inline ts/core/MathDocument.ts:278]: typo.

File: ts/core/MathDocument.ts [hunk 6]
```diff
@@ -254,178 +254,178 @@ export class RenderList<N, T, D> extends PrioritizedList<RenderData<N, T, D>> {
...
+   */
+  inputJax: InputJax<N, T, D>[];
+
+  /**
+   * The output jax to use for the document
+   */
+  outputJax: OutputJax<N, T, D>;
+
+  /**
+   * The DOM adaotor to use for input and output
```
Comment (zorkow) [inline ts/core/MathDocument.ts:300]: typo

File: ts/core/MathDocument.ts [hunk 7]
```diff
@@ -494,323 +497,364 @@ class DefaultMathItem<N, T, D> extends AbstractMathItem<N, T, D> {}
...
+      find:    [STATE.FINDMATH, 'findMath', '', false],
+      compile: [STATE.COMPILED],
+      metrics: [STATE.METRICS, 'getMetrics', '', false],
+      typeset: [STATE.TYPESET],
+      update:  [STATE.INSERTED, 'updateDocument', false]
+    }) as RenderActions<any, any, any>
+  };
+
+  /**
+   * A bit-field for the actions that heve been processed
```
Comment (zorkow) [inline ts/core/MathDocument.ts:530]: typo

File: ts/core/MathItem.ts [hunk 8]
```diff
@@ -80,142 +80,142 @@ export type BBox = {
...
-
-    /**
-     * Reset the item to its unprocessed state
-     *
-     * @param {number} restore  True if the original form should be restored
-     *                           to the document when rolling back a typeset version
-     */
-    reset(restore?: boolean): void;
+  /**
+   * The string represeting the expression to be processed
```
Comment (zorkow) [inline ts/core/MathItem.ts:84]: typo

File: ts/core/MathItem.ts [hunk 9]
```diff
@@ -80,142 +80,142 @@ export type BBox = {
...
+
+  /**
+   * The start and ending locations in the document of
+   *   this expression
+   */
+  start: Location<N, T>;
+  end: Location<N, T>;
+
+  /**
+   * The internal format for this expression (onece compiled)
```
Comment (zorkow) [inline ts/core/MathItem.ts:111]: typo

File: ts/core/MmlTree/Attributes.ts [hunk 10]
```diff
@@ -37,171 +37,183 @@ export const INHERIT = '_inherit_';
...
+   * @param {string} name  The name of a attribute to check
+   * @return {boolean}     True if attribute is set explicitly or inherited
+   *                         from an explicit mstyle or math attribute
+   */
+  public isSet(name: string): boolean {
+    return this.attributes.hasOwnProperty(name) || this.inherited.hasOwnProperty(name);
+  }
+
+  /**
+   * @param {string} name  The name of an attribute to test for the existance of a default
```
Comment (zorkow) [inline ts/core/MmlTree/Attributes.ts:156]: typo

File: ts/core/MmlTree/MmlNode.ts [hunk 11]
```diff
@@ -58,136 +58,140 @@ const TEXSPACELENGTH = ['', 'thinmathspace', 'mediummathspace', 'thickmathspace'
...
+  attributes: Attributes;
+
+  /**
+   * @return {MmlNode}  For embellished operators, the child node that contains the
+   *                    core <mo> node.  For non-embellished nodes, the original node.
+   */
+  core(): MmlNode;
+  /**
+   * @return {MmlNode}  For embellished operators, the core <mo> element (at whatever
+   *                    depth).  Fod non-embellished nodes, the original node itself.
```
Comment (zorkow) [inline ts/core/MmlTree/MmlNode.ts:132]: typo

File: ts/core/MmlTree/MmlNode.ts [hunk 12]
```diff
@@ -222,479 +226,497 @@ export interface MmlNodeClass extends NodeClass {
...
+
+  /**
+   * @return {boolean}  true if this is an inferred mrow
+   */
+  public get isInferred(): boolean {
+    return false;
+  }
+
+  /**
+   * @return {MmlNode}  The logial parent of this node (skipping over inferred rows
```
Comment (zorkow) [inline ts/core/MmlTree/MmlNode.ts:379]: typo

File: ts/core/MmlTree/MmlNode.ts [hunk 13]
```diff
@@ -222,479 +226,497 @@ export interface MmlNodeClass extends NodeClass {
...
+   * @param {bookean} prime             The TeX prime style to inherit (T vs. T', etc).
+   */
+  protected setChildInheritedAttributes(attributes: AttributeList, display: boolean, level: number, prime: boolean) {
+    for (const child of this.childNodes) {
+      child.setInheritedAttributes(attributes, display, level, prime);
+    }
+  }
+  /**
+   * Used by subclasses to add add their own attributes to the inherited list
+   * (e.g., mstyle uses this to augment the inhertied attibutes)
```
Comment (zorkow) [inline ts/core/MmlTree/MmlNode.ts:608]: typo

Comment (zorkow) [reply ts/core/MmlTree/MmlNode.ts:608]: Actually two

File: ts/core/MmlTree/MmlNode.ts [hunk 14]
```diff
@@ -900,177 +922,176 @@ export abstract class AbstractMmlBaseNode extends AbstractMmlNode {
...
+   *
+   * @override
+   */
+  public inheritAttributesFrom(_node: MmlNode) {}
+
+  /**
+   * No children or attributes, so ignore this call.
+   *
+   * @param {MmlNode} node          The node tree to be checked
+   * @param {PropertyList} options  The opritons for the check
```
Comment (zorkow) [inline ts/core/MmlTree/MmlNode.ts:1087]: typeo

File: ts/core/MmlTree/MmlNodes/munderover.ts [hunk 15]
```diff
@@ -30,121 +30,129 @@ import {AbstractMmlBaseNode, AttributeList} from '../MmlNode.js';
...
+   * @override
+   */
+  public get linebreakContainer() {
+    return true;
+  }
+
+  /**
+   * Base is in prime style if there is an over node
+   * Force scriptlevel change if converted to sub-sup by movablelimits on the base in non-display mode
+   * Adjust displaystyle, scriptlevel, and primestyle for the uncer/over nodes and check if accent
```
Comment (zorkow) [inline ts/core/MmlTree/MmlNodes/munderover.ts:96]: typo

**Pull request merged**